### PR TITLE
soundconverter: 4.0.3 -> 4.0.5

### PR DESCRIPTION
**Pull request by pyrox0**
## Description of changes

This upgrades the current soundconverter package to 4.0.5, as well as moves it to pkgs/by-name.
It's originally based on #259343, and the previous commit is there to preserve the work done. I've also tested that the package works, and I see no issues at all.

Per the comments on the original PR, I've removed 2 wrappers and solved the triple wrapper issue that blocked the previous one from being merged. I also added myself as a maintainer.

There's no formal changelog for the project, but you can see the changes after 4.0.3 at [this link](https://git.launchpad.net/soundconverter/log/?h=4.0.5) at the project's Launchpad repo.
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Aleksanaa** [comment]
The file is not moved but deleted and created. This will make it harder to view file history. You can try moving it in a separate commit. Other changes can be squashed into one commit.

**Comment by pyrox0** [comment]
Fixed that, thank you for noticing!

**Comment by pyrox0** [comment]
Ah, this won't evaluate cleanly until #306349 is merged. Once it is, I'll rebase and this should evaluate properly.

File: pkgs/by-name/so/soundconverter/package.nix [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+    makeWrapperArgs+=(--prefix PYTHONPATH : ${python3.pkgs.makePythonPath propagatedBuildInputs})
+  '';
+
+  # Necessary to set GDK_PIXBUF_MODULE_FILE.
+  strictDeps = false;
+
+  meta = with lib; {
```
Comment (eclairevoyant) [inline pkgs/by-name/so/soundconverter/package.nix:102]:
```suggestion
  meta = {
```

tracking issue: #292468

File: pkgs/by-name/so/soundconverter/package.nix [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    homepage = "https://soundconverter.org/";
+    description = "Leading audio file converter for the GNOME Desktop";
+    longDescription = ''
+      SoundConverter reads anything the GStreamer library can read,
+      and writes WAV, FLAC, MP3, AAC and Ogg Vorbis files.
+      Uses Python and GTK+ GUI toolkit, and runs on X Window System.
+    '';
+    license = licenses.gpl3Only;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [
```
Comment (eclairevoyant) [inline pkgs/by-name/so/soundconverter/package.nix:112]:
```suggestion
    license = lib.licenses.gpl3Only;
    platforms = lib.platforms.linux;
    maintainers = with lib.maintainers; [
```

File: pkgs/by-name/so/soundconverter/package.nix [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    xvfb-run python tests/test.py
+    runHook postCheck
+  '';
+
+  # Fixes https://ryantm.github.io/nixpkgs/languages-frameworks/gnome/#ssec-gnome-common-issues-double-wrapped
+  dontWrapGApps = true;
+
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+    makeWrapperArgs+=(--prefix PYTHONPATH : ${python3.pkgs.makePythonPath propagatedBuildInputs})
```
Comment (eclairevoyant) [inline pkgs/by-name/so/soundconverter/package.nix:96]:
```suggestion
    makeWrapperArgs+=(--prefix PYTHONPATH : ${python3.pkgs.makePythonPath propagatedBuildInputs})
```

why was this line needed?

File: pkgs/by-name/so/soundconverter/package.nix [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  fdk-aac-encoder,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "soundconverter";
+  version = "4.0.5";
+
+  src = fetchurl {
+    url = "https://launchpad.net/soundconverter/trunk/${version}/+download/${pname}-${version}.tar.gz";
+    sha256 = "1wvrxcvbhgqx9kliz5v2kvc4w08aw043v8ph4kgfbp2xd0v5mc3r";
```
Comment (eclairevoyant) [inline pkgs/by-name/so/soundconverter/package.nix:25]:
```suggestion
    hash = "...";
```

replacing ... with the SRI hash

**Review by eclairevoyant — COMMENTED** [review]
Thanks for the contribution! Some feedback below.
Also, for future reference, it's easier to review if reformatting is its own separate commit, so that formatting vs logical changes are separable.

File: pkgs/by-name/so/soundconverter/package.nix [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    python3Packages.gst-python
+    python3Packages.distutils-extra
+    python3Packages.setuptools
+    python3Packages.pygobject3
+  ];
+
+  nativeCheckInputs = [ xvfb-run ];
+
+  postPatch = ''
+    substituteInPlace  bin/soundconverter --replace \
```
[the remark this answers is not in the document]
Comment (pyrox0) [reply pkgs/by-name/so/soundconverter/package.nix:54]: Fixed

**Comment by pyrox0** [comment]
> Thanks for the contribution! Some feedback below. Also, for future reference, it's easier to review if reformatting is its own separate commit, so that formatting vs logical changes are separable.

I'll push an updated version with your suggestions and the formatting split out.

File: pkgs/by-name/so/soundconverter/package.nix (hunk 3 above)
Comment (pyrox0) [reply pkgs/by-name/so/soundconverter/package.nix:96]: This fixes an issue with PyGobject not initializing properly.

**Comment by eclairevoyant** [comment]
@ofborg eval

File: pkgs/by-name/so/soundconverter/package.nix [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+          self
+        ]
+      }
+      # FIXME: Fails due to weird Gio.file_parse_name() behavior.
+      sed -i '49 a\    @unittest.skip("Gio.file_parse_name issues")' tests/testcases/names.py
+    ''
+    + lib.optionalString (!faacSupport) ''
+      substituteInPlace tests/testcases/gui_integration.py --replace-warn \
+        "for encoder in ['fdkaacenc', 'faac', 'avenc_aac']:" \
+        "for encoder in ['fdkaacenc', 'avenc_aac']:"
```
Comment (SuperSandro2000) [inline pkgs/by-name/so/soundconverter/package.nix:81]:
```suggestion
      substituteInPlace tests/testcases/gui_integration.py --replace-fail \
        "for encoder in ['fdkaacenc', 'faac', 'avenc_aac']:" \
        "for encoder in ['fdkaacenc', 'avenc_aac']:"
```

File: pkgs/by-name/so/soundconverter/package.nix [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    python3Packages.gst-python
+    python3Packages.distutils-extra
+    python3Packages.setuptools
+    python3Packages.pygobject3
+  ];
+
+  nativeCheckInputs = [ xvfb-run ];
+
+  postPatch = ''
+    substituteInPlace  bin/soundconverter --replace-warn \
```
Comment (SuperSandro2000) [inline pkgs/by-name/so/soundconverter/package.nix:53]:
```suggestion
    substituteInPlace bin/soundconverter --replace-fail \
```
we want to hard fail if that doesn't match anymore

File: pkgs/by-name/so/soundconverter/package.nix [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+    (gst_all_1.gst-plugins-bad.override { inherit faacSupport; })
+  ];
+
+  nativeBuildInputs = [
+    intltool
+    wrapGAppsHook
+    gobject-introspection
+  ];
+
+  propagatedBuildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/by-name/so/soundconverter/package.nix:43]: If we wrap the executable anyway, do we even need to propogate those?

File: pkgs/by-name/so/soundconverter/package.nix (hunk 7 above)
Comment (pyrox0) [reply pkgs/by-name/so/soundconverter/package.nix]: Fixed

File: pkgs/by-name/so/soundconverter/package.nix (hunk 6 above)
Comment (pyrox0) [reply pkgs/by-name/so/soundconverter/package.nix:81]: Fixed

File: pkgs/by-name/so/soundconverter/package.nix (hunk 8 above)
Comment (pyrox0) [reply pkgs/by-name/so/soundconverter/package.nix]: Removed propagatedBuildInputs

**Review by jakubgs — APPROVED** [review]
Nice, thanks for the fixes.

File: pkgs/by-name/so/soundconverter/package.nix (hunk 6 above)
Comment (pyrox0) [reply pkgs/by-name/so/soundconverter/package.nix:82]: Actually updated this time.

File: pkgs/by-name/so/soundconverter/package.nix [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+  gsettings-desktop-schemas,
+  intltool,
+  xvfb-run,
+  gobject-introspection,
+  gst_all_1,
+  fdk-aac-encoder,
+}:
+let
+
+  pythonWrapPackages = [
```
Comment (Aleksanaa) [inline pkgs/by-name/so/soundconverter/package.nix:19]: This is how `dependencies` work. Can't understand why you have to use this dirty hack with `buildPythonApplication`.

**Review by Aleksanaa — COMMENTED** [review]
Also please squash commits into
1. migrate to `pkgs/by-name` (`soundconverter: migrate to pkgs/by-name`)
2. format (`soundconverter: format with nixfmt-rfc-style`)
3. update (`soundconverter: 4.0.3 -> 4.0.5`)

File: pkgs/by-name/so/soundconverter/package.nix (hunk 9 above)
Comment (Aleksanaa) [reply pkgs/by-name/so/soundconverter/package.nix:19]: See https://nixos.org/manual/nixpkgs/unstable/#reference

**Comment by JohnRTitor** [comment]
Hello @pyrox0 could you resolve the raised concerns?
